File: medium/leetcode98(验证二叉搜索树).java
public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode(int x) { val = x; }
}

class Solution {
    public boolean isValidBST(TreeNode root) {
        return helper(root, Long.MIN_VALUE, Long.MAX_VALUE);
    }
    public boolean helper(TreeNode root, Long min, Long max) {
        if (root == null) {
            return true;
        } 
        if ((root.val <= min) || (root.val >= max)) {
            return false;
        }
        return helper(root.left, min, (long) root.val) && helper(root.right, (long) root.val, max);
    }
}

File: medium/leetcode98(验证二叉搜索树).py
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution(object):
    def isValidBST(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        ansList = self.helper(root)
        for i in range(len(ansList)-1):
            if (ansList[i] >= ansList[i+1]):
                return False
        return True

    def helper(self, root):
        if root == None:
            return []
        else:
            return self.helper(root.left) + [root.val] + self.helper(root.right)